meshroom/ui/commands.py

```diff
@@ -350,6 +350,31 @@ def undoImpl(self):
         listAttribute.insert(self.index, self.value)
 
 
+class ClearImagesCommand(GraphCommand):
+    def __init__(self, graph, cameraInitNodes, parent=None):
+        super(ClearImagesCommand, self).__init__(graph, parent)
+        self.cameraInits = cameraInitNodes
+        self.viewpoints = [cameraInit.attribute("viewpoints").getExportValue() for cameraInit in self.cameraInits]
+        self.intrinsics = [cameraInit.attribute("intrinsics").getExportValue() for cameraInit in self.cameraInits]
+        self.title = "Clear{}Images".format(" " if len(self.cameraInits) == 1 else " All ")
+        self.setText(self.title)
+
+    def redoImpl(self):
+        for i in range(len(self.cameraInits)):
+            # Reset viewpoints
+            self.cameraInits[i].viewpoints.resetValue()
+            self.cameraInits[i].viewpoints.valueChanged.emit()
+            # Reset intrinsics
+            self.cameraInits[i].intrinsics.resetValue()
+            self.cameraInits[i].intrinsics.valueChanged.emit()
+
+    def undoImpl(self):
+        for i in range(len(self.cameraInits)):
+            with GraphModification(self.graph):
+                self.cameraInits[i].viewpoints.value = self.viewpoints[i]
```

Comment:
Sounds dangerous to rely on the ordering of cameraInit nodes.
It would be more straightforward to use a map with the unique name of the node as key.

Reply:
Done.